src/checker.rs

```diff
@@ -0,0 +1,88 @@
+use std::error::Error;
+use std::time::Duration;
+
+use reqwest::{ClientBuilder, StatusCode};
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub enum CheckResult {
+    SUCCESS,
+    FAILURE(FailureReason),
+    MISSED,
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub enum FailureReason {
+    Timeout,
+    DnsError(String),
+    ConfigError(String),
+    Error(String),
+}
+
+pub async fn check_domain(domain: String) -> CheckResult {
+    // TODO: Timeout should come from config
+    let timeout = Duration::new(5, 0);
```

Comment:
I think `from_secs` is clearer.

```suggestion
    let timeout = Duration::from_secs(5);
```